app/models/robots/robot.rb

```diff
@@ -74,18 +75,35 @@ def bed_plates=(bed_plates)
     private
 
     def valid_robot
-      if verify_robot? && beds.values.first.plate.present?
-        if beds.values.first.plate.custom_metadatum_collection.nil?
-          error_messages << 'Your plate is not on the right robot'
-          return false
-        elsif beds.values.first.plate.custom_metadatum_collection.metadata['created_with_robot'] != robot_barcode
-          error_messages << 'Your plate is not on the right robot'
-          return false
-        end
+      return false unless robot_present_if_required
+
+      return true unless verify_robot? && beds.values.first.plate.present?
+
+      if missing_custom_metadatum_collection || original_robot != robot_barcode
```

Comment:
Does this line work ok if original robot is not set? e.g. nil != '1234'

Reply:
It's equivalent to what was there before isn't it?

I can see a case that isn't explicitly dealt with where missing_custom_metadatum_collection is false, but there isn't one with the key 'created_with_robot' - looks like that would end up with nil != '1234', so add an error message, which I think is probably correct?

Reply:
Yep it's fine, the guard clause I was confused about above stops it getting there unless we want the robot to be the same. Ignore me.